diff --git a/engines/scumm/charset.cpp b/engines/scumm/charset.cpp
index 73685802489..42b298f5c30 100644
--- a/engines/scumm/charset.cpp
+++ b/engines/scumm/charset.cpp
@@ -236,7 +236,7 @@ void ScummEngine::loadKorFont() {
 }
 
 byte *ScummEngine::get2byteCharPtr(int idx) {
-	if (_game.platform == Common::kPlatformFMTowns || _game.platform == Common::kPlatformPCEngine)
+	if (!isScummvmKorTarget() && (_game.platform == Common::kPlatformFMTowns || _game.platform == Common::kPlatformPCEngine))
 		return nullptr;
 
 	switch (_language) {
@@ -513,7 +513,7 @@ int CharsetRenderer::getStringWidth(int arg, const byte *text) {
 		}
 
 		if (_vm->_useCJKMode) {
-			if (_vm->_game.platform == Common::kPlatformFMTowns) {
+			if (_vm->_language == Common::JA_JPN && _vm->_game.platform == Common::kPlatformFMTowns) {
 				if (checkSJISCode(chr))
 					// This strange character conversion is the exact way the original does it here.
 					// This is the only way to get an accurate text formatting in the MI1 intro.
@@ -606,7 +606,7 @@ void CharsetRenderer::addLinebreaks(int a, byte *str, int pos, int maxwidth) {
 			lastspace = pos - 1;
 
 		if (_vm->_useCJKMode) {
-			if (_vm->_game.platform == Common::kPlatformFMTowns) {
+			if (_vm->_language == Common::JA_JPN && _vm->_game.platform == Common::kPlatformFMTowns) {
 				if (checkSJISCode(chr))
 					// This strange character conversion is the exact way the original does it here.
 					// This is the only way to get an accurate text formatting in the MI1 intro.
@@ -758,6 +758,9 @@ void CharsetRendererPC::drawBits1Kor(Graphics::Surface &dest, int x1, int y1, co
 	default: // Stroke
 		i = 5;
 	}
+	// check!!
+	//if (!_enableShadow)
+	//	i = 13;
 
 	const byte *origSrc = src;
 	byte *origDst = dst;
@@ -1439,7 +1442,7 @@ int CharsetRendererTownsV3::getDrawHeightIntern(uint16 chr) {
 }
 
 void CharsetRendererTownsV3::setDrawCharIntern(uint16 chr) {
-	_sjisCurChar = (_vm->_useCJKMode && chr > 127) ? chr : 0;
+	_sjisCurChar = (!_vm->isScummvmKorTarget() && _vm->_useCJKMode && chr > 127) ? chr : 0;
 }
 #endif
 
