diff --git a/input/drivers_joypad/udev_joypad.c b/input/drivers_joypad/udev_joypad.c
index a2db95e49a..606810ec47 100644
--- a/input/drivers_joypad/udev_joypad.c
+++ b/input/drivers_joypad/udev_joypad.c
@@ -20,6 +20,7 @@
 #include <limits.h>
 #include <fcntl.h>
 #include <errno.h>
+#include <stdio.h>
 
 #include <sys/types.h>
 #include <sys/stat.h>
@@ -99,6 +100,27 @@ static struct udev *udev_joypad_fd             = NULL;
 static struct udev_monitor *udev_joypad_mon    = NULL;
 static struct udev_joypad udev_pads[MAX_USERS];
 
+static int udev_init_rg351p_rumble(void)
+{
+   FILE *fp;
+   fp = fopen("/sys/class/pwm/pwmchip0/pwm0/period", "w");
+   fprintf(fp, "1000000");
+   fclose(fp);
+   fp = fopen("/sys/class/pwm/pwmchip0/pwm0/enable", "w");
+   fprintf(fp, "1");
+   fclose(fp);
+   fp = fopen("/sys/class/pwm/pwmchip0/pwm0/duty_cycle", "w");
+   fprintf(fp, "1000000");
+   fclose(fp);
+}
+
+static int udev_set_rg351p_rumble(bool on)
+{
+   FILE *fp = fopen("/sys/class/pwm/pwmchip0/pwm0/duty_cycle", "w");
+   fprintf(fp, on ? "10" : "1000000");
+   fclose(fp);
+}
+
 static INLINE int16_t udev_compute_axis(const struct input_absinfo *info, int value)
 {
    int range = info->maximum - info->minimum;
@@ -266,6 +288,13 @@ static int udev_add_pad(struct udev_device *dev, unsigned p, int fd, const char
                p, path, pad->num_effects);
    }
 
+   /* Initialize RG351P rumble */
+   if (pad->vid == 0x1209 && pad->pid == 0x3100)
+   {
+	  udev_init_rg351p_rumble();
+      pad->num_effects = 2;
+   }
+
    return ret;
 }
 
@@ -317,6 +346,10 @@ static void udev_check_device(struct udev_device *dev, const char *path)
 
 static void udev_free_pad(unsigned pad)
 {
+   /* Turn off RG351P rumble */
+   if (udev_pads[pad].vid == 0x1209 && udev_pads[pad].pid == 0x3100)
+	  udev_set_rg351p_rumble(false);
+
    if (udev_pads[pad].fd >= 0)
       close(udev_pads[pad].fd);
 
@@ -376,6 +409,18 @@ static bool udev_set_rumble(unsigned i,
    if (pad->num_effects < 2)
       return false;
 
+   /* Set RG351P rumble */
+   if (pad->vid == 0x1209 && pad->pid == 0x3100) {
+	  if (effect == RETRO_RUMBLE_STRONG) { 
+         old_strength = pad->strength[effect];
+         if (old_strength == strength)
+            return true;
+         udev_set_rg351p_rumble(strength > 0 ? true : false);
+         pad->strength[effect] = strength;
+      }
+      return true;
+   }
+
    old_strength = pad->strength[effect];
    if (old_strength == strength)
       return true;
