diff --git a/deps/libgo2/src/display.c b/deps/libgo2/src/display.c
index b91cf0624b..f6d681266a 100644
--- a/deps/libgo2/src/display.c
+++ b/deps/libgo2/src/display.c
@@ -434,7 +434,7 @@ go2_surface_t* go2_surface_create(go2_display_t* display, int width, int height,
         goto out;
     }
 
-
+    c_RkRgaInit();	// british.choi : OGA black screen
     result->display = display;
     result->gem_handle = args.handle;
     result->size = args.size;
@@ -456,6 +456,7 @@ void go2_surface_destroy(go2_surface_t* surface)
     struct drm_mode_destroy_dumb args = { 0 };
     args.handle = surface->gem_handle;
 
+    c_RkRgaDeInit();	// british.choi : OGA black screen
     int io = drmIoctl(surface->display->fd, DRM_IOCTL_MODE_DESTROY_DUMB, &args);
     if (io < 0)
     {
@@ -909,7 +910,7 @@ go2_presenter_t* go2_presenter_create(go2_display_t* display, uint32_t format, u
 
     memset(result, 0, sizeof(*result));
 
-
+    c_RkRgaInit();	// british.choi : OGA black screen
     result->display = display;
     result->format = format;
     result->background_color = background_color;
@@ -977,6 +978,7 @@ void go2_presenter_destroy(go2_presenter_t* presenter)
         }
     }
 
+    c_RkRgaDeInit();	// british.choi : OGA black screen
     free(presenter);
 }
 
diff --git a/gfx/drivers/oga_gfx.c b/gfx/drivers/oga_gfx.c
index 66680d3c52..2e064635ae 100644
--- a/gfx/drivers/oga_gfx.c
+++ b/gfx/drivers/oga_gfx.c
@@ -224,7 +224,7 @@ oga_surface_t* oga_create_surface(int display_fd,
       RARCH_ERR("DRM_IOCTL_MODE_CREATE_DUMB failed.\n");
       goto out;
    }
-
+
    surface->display_fd = display_fd;
    surface->handle     = args.handle;
    surface->width      = width;
@@ -245,6 +245,7 @@ oga_surface_t* oga_create_surface(int display_fd,
       return NULL;
    }
 
+   c_RkRgaInit();	// british.choi : OGA black screen
    return surface;
 
 out:
@@ -259,6 +259,7 @@ void oga_destroy_surface(oga_surface_t* surface)
 
    args.handle = surface->handle;
 
+   c_RkRgaDeInit();	// british.choi : OGA black screen
    io          = drmIoctl(surface->display_fd,
          DRM_IOCTL_MODE_DESTROY_DUMB, &args);
    if (io < 0)
