diff --git a/src/joystick/SDL_joystick.c b/src/joystick/SDL_joystick.c
index ba4f819..2a18c94 100644
--- a/src/joystick/SDL_joystick.c
+++ b/src/joystick/SDL_joystick.c
@@ -1686,7 +1686,7 @@ SDL_JoystickUpdate(void)
             if (joystick->delayed_guide_button) {
                 SDL_GameControllerHandleDelayedGuideButton(joystick);
             }
-        }
+
 
         if (joystick->rumble_expiration) {
             SDL_LockJoysticks();
@@ -1707,6 +1707,8 @@ SDL_JoystickUpdate(void)
             }
             SDL_UnlockJoysticks();
         }
+
+	}
     }
 
     SDL_LockJoysticks();
