diff --git a/src/joystick/SDL_joystick.c b/src/joystick/SDL_joystick.c
index 724239e..8058947 100644
--- a/src/joystick/SDL_joystick.c
+++ b/src/joystick/SDL_joystick.c
@@ -1554,7 +1554,7 @@ SDL_JoystickUpdate(void)
             if (joystick->delayed_guide_button) {
                 SDL_GameControllerHandleDelayedGuideButton(joystick);
             }
-        }
+
 
         if (joystick->rumble_expiration) {
             SDL_LockJoysticks();
@@ -1575,6 +1575,8 @@ SDL_JoystickUpdate(void)
             }
             SDL_UnlockJoysticks();
         }
+
+	}
     }
 
     SDL_LockJoysticks();
