--- a/src/burner/sdl/drv.cpp	2022-01-22 01:27:13.278967632 -0600
+++ b/src/burner/sdl/drv.cpp	2022-01-22 01:24:57.509794634 -0600
@@ -3,7 +3,7 @@
 #include "neocdlist.h"
 int bDrvOkay = 0;                       // 1 if the Driver has been initted okay, and it's okay to use the BurnDrv functions
 
-char szAppRomPaths[DIRS_MAX][MAX_PATH] = { { "/usr/local/share/roms/" }, { "roms/" }, };
+char szAppRomPaths[DIRS_MAX][MAX_PATH] = { { "/storage/roms/fbneo/" }, { "/storage/roms/bios/" }, };
 
 static bool bSaveRAM = false;
 
diff --git a/src/dep/scripts/gamelist.pl b/src/dep/scripts/gamelist.pl
index d476d1010..20fb6e8b6 100755
--- a/src/dep/scripts/gamelist.pl
+++ b/src/dep/scripts/gamelist.pl
@@ -327,7 +327,7 @@ print OUTFILE << "CPPEND";
 // Lookup table containing sourcefiles of all drivers
 struct game_sourcefile_entry {
 	char game_name[32];
-	char sourcefile[32];
+	char sourcefile[256];
 };
 
 static game_sourcefile_entry sourcefile_table[] = {
