--- a/board/rockchip/odroidgoa/odroidgoa.c
+++ b/board/rockchip/odroidgoa/odroidgoa.c
@@ -143,10 +143,10 @@
 	}
 
 	/* check launcher in ext4 fs of sd card */
-	if (file_exists("mmc", "1:2", "/usr/local/bin/emulationstation/emulationstation",
+	if (!file_exists("mmc", "1:2", "/usr/local/bin/emulationstation/emulationstation",
 				FS_TYPE_EXT)) {
 		lcd_setfg_color("white");
-		lcd_printf(0, 0, 1, "[ GO Advanced EMULATION Image ]");
+		lcd_printf(0, 0, 1, "[ EmuELEC RG351P ]");
 	}
 
 	return;
diff --git a/board/rockchip/odroidgoa/odroidgoa_power.c b/board/rockchip/odroidgoa/odroidgoa_power.c
index dce99e45fb..cd7b8693d5 100644
--- a/board/rockchip/odroidgoa/odroidgoa_power.c
+++ b/board/rockchip/odroidgoa/odroidgoa_power.c
@@ -14,7 +14,7 @@
 
 #define PWR_LED_GPIO	18	/* GPIO0_C2 */
 #define DC_DET_GPIO	11	/* GPIO0_B3 */
-#define CHG_LED_GPIO	13	/* GPIO0_B5 */
+#define CHG_LED_GPIO	77	/* GPIO2_B5 */
 
 void board_chg_led(void)
 {
diff --git a/cmd/hwrev.c b/cmd/hwrev.c
index 640397e1cb..6e50cc364c 100644
--- a/cmd/hwrev.c
+++ b/cmd/hwrev.c
@@ -22,7 +22,7 @@ int do_hwrev(cmd_tbl_t *cmdtp, int flag, int argc, char * const argv[])
 	/* GO2 rev 1.1 */
 	if (check_range(655, 695, hwrev_adc)) {
 		env_set("hwrev", "v11");
-		env_set("dtb_name", "rk3326-odroidgo2-linux-v11.dtb");
+		env_set("dtb_name", "rk3326-rg351p-linux.dtb");
 	}
 	/* GO2 rev 1.0 */
 	else if (check_range(816, 896, hwrev_adc)) {
diff --git a/cmd/odroidtest.c b/cmd/odroidtest.c
index 10dc550b0c..5e800ccf02 100644
--- a/cmd/odroidtest.c
+++ b/cmd/odroidtest.c
@@ -801,7 +801,7 @@ static int do_odroidtest_btn(cmd_tbl_t * cmdtp, int flag,
 
 #define PWR_LED_GPIO	18	/* GPIO0_C2 */
 #define DC_DET_GPIO	11	/* GPIO0_B3 */
-#define CHG_LED_GPIO	13	/* GPIO0_B5 */
+#define CHG_LED_GPIO	77	/* GPIO2_B5 */
 static int do_odroidtest_bat(cmd_tbl_t * cmdtp, int flag,
 				int argc, char * const argv[])
 {
