diff --git a/files/lang/Makefile b/files/lang/Makefile
index 97eea7d5..a40df128 100644
--- a/files/lang/Makefile
+++ b/files/lang/Makefile
@@ -31,14 +31,45 @@ merge:
 # Spanish: drop accents transliterated with `"` (which breaks translation file format)
 # and transliterate the rest using the es_ES.UTF-8 locale
 es.mo: es.po
-	sed -e 'y/äëïöőüűÄËÏŐÖÜŰ/aeioouuAEIOOUU/' $< > $<.tmp
+	#sed -e 'y/äëïöőüűÄËÏŐÖÜŰ/aeioouuAEIOOUU/' $< > $<.tmp
+	sed -e 'y/äëïöőüűÄËÏŐÖÜŰ/äëïöőüűÄËÏŐÖÜŰ/' $< > $<.tmp
+	sed -i -e 's/ä/a/g' $<.tmp
+	sed -i -e 's/ë/e/g' $<.tmp
+	sed -i -e 's/ï/i/g' $<.tmp
+	sed -i -e 's/ö/o/g' $<.tmp
+	sed -i -e 's/ő/o/g' $<.tmp
+	sed -i -e 's/ü/u/g' $<.tmp
+	sed -i -e 's/ű/u/g' $<.tmp
+	sed -i -e 's/Ä/A/g' $<.tmp
+	sed -i -e 's/Ë/E/g' $<.tmp
+	sed -i -e 's/Ï/I/g' $<.tmp
+	sed -i -e 's/Ő/O/g' $<.tmp
+	sed -i -e 's/Ö/O/g' $<.tmp
+	sed -i -e 's/Ü/U/g' $<.tmp
+	sed -i -e 's/Ű/U/g' $<.tmp
 	LANG=es_ES.UTF-8 LC_ALL=es_ES.UTF-8 LC_CTYPE=es_ES.UTF-8 iconv -f utf-8 -t ascii//TRANSLIT $<.tmp > $<.2.tmp
 	msgfmt $<.2.tmp -o $@
 
 # In French, accented characters are mapped to unused ASCII characters
 # Non-mapped characters are replaced by their non-accented equivalents
 fr.mo: fr.po
-	sed -e 'y/àâçéèêîïôùûüÉÊÀ/@*^~`|><#&$$uEEA/' $< > $<.tmp
+	#sed -e 'y/àâçéèêîïôùûüÉÊÀ/@*^~`|><#&$$uEEA/' $< > $<.tmp
+	sed -e 'y/àâçéèêîïôùûüÉÊÀ/àâçéèêîïôùûüÉÊÀ/' $< > $<.tmp
+	sed -i -e 's/à/@/g' $<.tmp
+	sed -i -e 's/â/*/g' $<.tmp
+	sed -i -e 's/ç/^/g' $<.tmp
+	sed -i -e 's/é/~/g' $<.tmp
+	sed -i -e 's/è/`/g' $<.tmp
+	sed -i -e 's/ê/|/g' $<.tmp
+	sed -i -e 's/î/>/g' $<.tmp
+	sed -i -e 's/ï/</g' $<.tmp
+	sed -i -e 's/ô/#/g' $<.tmp
+	sed -i -e 's/ù/&/g' $<.tmp
+	sed -i -e 's/û/$$/g' $<.tmp
+	sed -i -e 's/ü/u/g' $<.tmp
+	sed -i -e 's/É/E/g' $<.tmp
+	sed -i -e 's/Ê/E/g' $<.tmp
+	sed -i -e 's/À/A/g' $<.tmp
 	sed -i~ -e 's/œ/oe/g' $<.tmp
 	msgfmt $<.tmp -o $@
 
@@ -83,7 +114,22 @@ it.mo: it.po
 # All other languages: drop accents transliterated with `"` (which breaks translation file format)
 # and transliterate the rest with default iconv rules
 %.mo: %.po
-	sed -e 'y/äëïöőüűÄËÏŐÖÜŰ/aeioouuAEIOOUU/' $< > $<.tmp
+	#sed -e 'y/äëïöőüűÄËÏŐÖÜŰ/aeioouuAEIOOUU/' $< > $<.tmp
+	sed -e 'y/äëïöőüűÄËÏŐÖÜŰ/äëïöőüűÄËÏŐÖÜŰ/' $< > $<.tmp
+	sed -i -e 's/ä/a/g' $<.tmp
+	sed -i -e 's/ë/e/g' $<.tmp
+	sed -i -e 's/ï/i/g' $<.tmp
+	sed -i -e 's/ö/o/g' $<.tmp
+	sed -i -e 's/ő/o/g' $<.tmp
+	sed -i -e 's/ü/u/g' $<.tmp
+	sed -i -e 's/ű/u/g' $<.tmp
+	sed -i -e 's/Ä/A/g' $<.tmp
+	sed -i -e 's/Ë/E/g' $<.tmp
+	sed -i -e 's/Ï/I/g' $<.tmp
+	sed -i -e 's/Ő/O/g' $<.tmp
+	sed -i -e 's/Ö/O/g' $<.tmp
+	sed -i -e 's/Ü/U/g' $<.tmp
+	sed -i -e 's/Ű/U/g' $<.tmp
 	iconv -f utf-8 -t ascii//TRANSLIT $<.tmp > $<.2.tmp
 	msgfmt $<.2.tmp -o $@
 
