diff --git a/modules/libjoy/libjoy.c b/modules/libjoy/libjoy.c
index 0350d31..d8c599d 100644
--- a/modules/libjoy/libjoy.c
+++ b/modules/libjoy/libjoy.c
@@ -186,6 +186,15 @@ int libjoy_get_position( int axis )
 {
     if ( _selected_joystick >= 0 && _selected_joystick < _max_joys )
     {
+        if (axis == SDL_CONTROLLER_AXIS_LEFTX || axis == SDL_CONTROLLER_AXIS_LEFTY)
+        {
+            const char *name = SDL_JoystickName(_joysticks[ _selected_joystick ]);
+            if (name && SDL_strstr(name, "OpenSimHardware OSH PB Controller"))
+            {
+                /* Reverse Axis X, Y */
+                return SDL_JoystickGetAxis( _joysticks[ _selected_joystick ], axis ) * -1 ;
+            }
+        }
         return SDL_JoystickGetAxis( _joysticks[ _selected_joystick ], axis ) ;
     }
     return 0 ;
@@ -368,6 +377,15 @@ int libjoy_get_position_specific( int joy, int axis )
     {
         if ( axis >= 0 && axis <= SDL_JoystickNumAxes( _joysticks[ joy ] ) )
         {
+            if (axis == SDL_CONTROLLER_AXIS_LEFTX || axis == SDL_CONTROLLER_AXIS_LEFTY)
+            {
+                const char *name = SDL_JoystickName(_joysticks[ joy ]);
+                if (name && SDL_strstr(name, "OpenSimHardware OSH PB Controller"))
+                {
+                    /* Reverse Axis X, Y */
+                    return SDL_JoystickGetAxis( _joysticks[ joy ], axis ) * -1 ;
+                }
+            }
             return SDL_JoystickGetAxis( _joysticks[ joy ], axis ) ;
         }
     }
