diff --git a/build/Makefile b/build/Makefile
index f0c9fd0..0a95c13 100755
--- a/build/Makefile
+++ b/build/Makefile
@@ -317,7 +317,7 @@ else ifneq (,$(findstring oga_a35_neon_hardfloat,$(platform)))
    USE_THEORADEC = 0
    USE_TREMOR = 1
    HAVE_MT32EMU = 0
-   DEFINES +=  -marm -mtune=cortex-a35 -mfpu=neon-fp-armv8 -mfloat-abi=hard -march=armv8-a+crc
+   DEFINES +=  -marm -mtune=cortex-a35 -mfpu=neon-fp-armv8 -mfloat-abi=hard -march=armv8-a -Ofast -flto -fuse-linker-plugin -fno-stack-protector -fno-ident -fomit-frame-pointer -fmerge-all-constants -ffast-math -funroll-all-loops
    HAVE_NEON = 1
    DEFINES += -DARM
 
