--- a/cheevos/cheevos.c
+++ b/cheevos/cheevos.c
@@ -329,6 +329,12 @@
 
    CHEEVOS_LOG(RCHEEVOS_TAG "Awarding achievement %u: %s (%s)\n",
          cheevo->id, cheevo->title, cheevo->description);
+
+    char pixelcade[256];
+    snprintf(pixelcade, sizeof(pixelcade), "/storage/.emulationstation/scripts/achievements/pixelcade.sh %u \"%s\" \"%s\" > /dev/null 2>/dev/null",
+    cheevo->id, cheevo->title, cheevo->description);
+    //CHEEVOS_LOG(pixelcade);
+    system(pixelcade);
 
    /* Show the on screen message. */
 #if defined(HAVE_GFX_WIDGETS)
