--- a/tasks/task_content.c	2019-01-15 12:55:14.000000000 -0600
+++ b/tasks/task_content.c	2019-01-28 22:00:41.340468631 -0600
@@ -500,6 +500,7 @@
    {
       RARCH_LOG("[CORE]: Using content: %s.\n", args->content_path);
       argv[(*argc)++] = strdup(args->content_path);
+      system("/usr/bin/fbfix"); // emuelec
    }
 #ifdef HAVE_MENU
    else
@@ -726,6 +727,8 @@
    }
 
    *buf = ret_buf;
+    
+    system("/usr/bin/fbfix"); // emuelec
 
    return true;
 }
@@ -1064,7 +1067,7 @@
    else
       rcheevos_pause_hardcore();
 #endif
-
+   system("/usr/bin/fbfix"); // emuelec
    return true;
 }

--- a/retroarch.c	2019-01-15 12:55:14.000000000 -0600
+++ b/retroarch.c	2019-01-28 22:00:41.340468631 -0600 
@@ -12409,6 +12409,7 @@
       )
       input_remapping_set_defaults(true);
 #endif
+system("/usr/bin/fbfix"); // emuelec
 }
 
 #ifdef HAVE_CHEATS
