--- a/tasks/task_content.c
+++ b/tasks/task_content.c
@@ -521,6 +521,7 @@ static void content_load_init_wrap(
    {
       RARCH_LOG("Using content: %s.\n", args->content_path);
       argv[(*argc)++] = strdup(args->content_path);
+      system("/storage/.config/emuelec/bin/fbfix"); // emuelec
    }
 #ifdef HAVE_MENU
    else
@@ -713,6 +714,8 @@ static bool load_content_into_memory(

    *buf = ret_buf;

+   system("/storage/.config/emuelec/bin/fbfix"); // emuelec
+
    return true;
 }

@@ -1087,6 +1090,7 @@ static bool content_file_load(
       rcheevos_hardcore_paused = true;
 #endif

+   system("/storage/.config/emuelec/bin/fbfix"); // emuelec
    return true;
 }


--- a/retroarch.c
+++ b/retroarch.c
@@ -6508,6 +6508,7 @@ static void command_event_deinit_core(bool reinit)
       )
       input_remapping_set_defaults(true);
 #endif
+   system("/storage/.config/emuelec/bin/fbfix"); // emuelec
 }

 static void command_event_init_cheats(void)
