diff --git a/Source/controls/plrctrls.cpp b/Source/controls/plrctrls.cpp
index 69455f94..c385b633 100644
--- a/Source/controls/plrctrls.cpp
+++ b/Source/controls/plrctrls.cpp
@@ -738,6 +738,7 @@ void ResetInvCursorPosition()
 	mousePos.y -= (InventorySlotSizeInPixels.height / 2);
 
 	SetCursorPos(mousePos);
+	MousePosition = mousePos;
 }
 
 int FindClosestInventorySlot(Point mousePos)
@@ -1033,6 +1034,7 @@ void InventoryMove(AxisDirection dir)
 	}
 
 	SetCursorPos(mousePos);
+	MousePosition = mousePos;
 }
 
 /**
@@ -1621,6 +1623,7 @@ void HandleRightStickMotion()
 
 			ControlMode = ControlTypes::KeyboardAndMouse;
 			lastMouseSetTick = now;
+			MousePosition = { x, y };
 		}
 	}
 }
