diff --git a/Source/controls/devices/game_controller.cpp b/Source/controls/devices/game_controller.cpp
index 1faf94ba..61b69f10 100644
--- a/Source/controls/devices/game_controller.cpp
+++ b/Source/controls/devices/game_controller.cpp
@@ -50,13 +50,13 @@ ControllerButton GameController::ToControllerButton(const SDL_Event &event)
 	case SDL_CONTROLLERBUTTONUP:
 		switch (event.cbutton.button) {
 		case SDL_CONTROLLER_BUTTON_A:
-			return ControllerButton_BUTTON_A;
-		case SDL_CONTROLLER_BUTTON_B:
 			return ControllerButton_BUTTON_B;
+		case SDL_CONTROLLER_BUTTON_B:
+			return ControllerButton_BUTTON_A;
 		case SDL_CONTROLLER_BUTTON_X:
-			return ControllerButton_BUTTON_X;
-		case SDL_CONTROLLER_BUTTON_Y:
 			return ControllerButton_BUTTON_Y;
+		case SDL_CONTROLLER_BUTTON_Y:
+			return ControllerButton_BUTTON_X;
 		case SDL_CONTROLLER_BUTTON_LEFTSTICK:
 			return ControllerButton_BUTTON_LEFTSTICK;
 		case SDL_CONTROLLER_BUTTON_RIGHTSTICK:
@@ -92,13 +92,13 @@ SDL_GameControllerButton GameController::ToSdlGameControllerButton(ControllerBut
 		UNIMPLEMENTED();
 	switch (button) {
 	case ControllerButton_BUTTON_A:
-		return SDL_CONTROLLER_BUTTON_A;
-	case ControllerButton_BUTTON_B:
 		return SDL_CONTROLLER_BUTTON_B;
+	case ControllerButton_BUTTON_B:
+		return SDL_CONTROLLER_BUTTON_A;
 	case ControllerButton_BUTTON_X:
-		return SDL_CONTROLLER_BUTTON_X;
-	case ControllerButton_BUTTON_Y:
 		return SDL_CONTROLLER_BUTTON_Y;
+	case ControllerButton_BUTTON_Y:
+		return SDL_CONTROLLER_BUTTON_X;
 	case ControllerButton_BUTTON_BACK:
 		return SDL_CONTROLLER_BUTTON_BACK;
 	case ControllerButton_BUTTON_START:
