diff --git a/Source/options.cpp b/Source/options.cpp
index 66f271bc..3eb2d9bf 100644
--- a/Source/options.cpp
+++ b/Source/options.cpp
@@ -316,7 +316,7 @@ void LoadOptions()
 	sgOptions.Graphics.bVSync = GetIniBool("Graphics", "Vertical Sync", true);
 	sgOptions.Graphics.nGammaCorrection = GetIniInt("Graphics", "Gamma Correction", 100);
 #if SDL_VERSION_ATLEAST(2, 0, 0)
-	sgOptions.Graphics.bHardwareCursor = GetIniBool("Graphics", "Hardware Cursor", HardwareCursorDefault());
+	sgOptions.Graphics.bHardwareCursor = GetIniBool("Graphics", "Hardware Cursor", false);
 	sgOptions.Graphics.bHardwareCursorForItems = GetIniBool("Graphics", "Hardware Cursor For Items", false);
 	sgOptions.Graphics.nHardwareCursorMaxSize = GetIniInt("Graphics", "Hardware Cursor Maximum Size", 128);
 #endif
@@ -794,7 +794,7 @@ void OptionEntryLanguageCode::LoadFromIni(string_view category)
 	if (!HasTranslation(locale)) {
 		locale = locale.substr(0, 2);
 		if (!HasTranslation(locale)) {
-			locale = "en";
+			locale = "ko_KR";
 		}
 	}
 	LogVerbose("Best match locale: {}", locale);
