diff --git a/Source/options.cpp b/Source/options.cpp
index 9e03df04..119d789e 100644
--- a/Source/options.cpp
+++ b/Source/options.cpp
@@ -1206,7 +1206,7 @@ void OptionEntryLanguageCode::LoadFromIni(string_view category)
 	}
 
 	LogVerbose("No suitable translation found");
-	strcpy(szCode, "en");
+	strcpy(szCode, "ko");	// default language to ko
 }
 void OptionEntryLanguageCode::SaveToIni(string_view category) const
 {
@@ -1231,7 +1231,8 @@ void OptionEntryLanguageCode::CheckLanguagesAreInitialized() const
 	languages.emplace_back("it", "Italiano");
 
 	if (HaveExtraFonts()) {
-		languages.emplace_back("ja", "日本語");
+		//remove japanese font
+		//languages.emplace_back("ja", "日本語");
 		languages.emplace_back("ko", "한국어");
 	}
 
@@ -1243,8 +1244,9 @@ void OptionEntryLanguageCode::CheckLanguagesAreInitialized() const
 	languages.emplace_back("uk", "Українська");
 
 	if (HaveExtraFonts()) {
-		languages.emplace_back("zh_CN", "汉语");
-		languages.emplace_back("zh_TW", "漢語");
+		//remove chinese font
+		//languages.emplace_back("zh_CN", "汉语");
+		//languages.emplace_back("zh_TW", "漢語");
 	}
 
 	// Ensures that the ini specified language is present in languages list even if unknown (for example if someone starts to translate a new language)
