diff --git a/Source/options.cpp b/Source/options.cpp
index 2982587a..4426bf45 100644
--- a/Source/options.cpp
+++ b/Source/options.cpp
@@ -221,7 +221,7 @@ void SaveIni()
 #if SDL_VERSION_ATLEAST(2, 0, 0)
 bool HardwareCursorDefault()
 {
-#if defined(__ANDROID__) || defined(TARGET_OS_IPHONE)
+#if 1//#if defined(__ANDROID__) || defined(TARGET_OS_IPHONE)
 	// See https://github.com/diasurgical/devilutionX/issues/2502
 	return false;
 #else
@@ -306,7 +306,7 @@ Options sgOptions;
 #if SDL_VERSION_ATLEAST(2, 0, 0)
 bool HardwareCursorSupported()
 {
-#if defined(TARGET_OS_IPHONE)
+#if 1//#if defined(TARGET_OS_IPHONE)
 	return false;
 #else
 	SDL_version v;
@@ -953,7 +953,7 @@ void OptionEntryLanguageCode::LoadFromIni(string_view category)
 	}
 
 	LogVerbose("No suitable translation found");
-	strcpy(szCode, "en");
+	strcpy(szCode, "ko");	// default language to ko
 }
 void OptionEntryLanguageCode::SaveToIni(string_view category) const
 {
@@ -977,7 +977,8 @@ void OptionEntryLanguageCode::CheckLanguagesAreInitialized() const
 	languages.emplace_back("it", "Italiano");
 
 	if (font_mpq) {
-		languages.emplace_back("ja", "日本語");
+		//remove japanese font
+		//languages.emplace_back("ja", "日本語");
 		languages.emplace_back("ko", "한국어");
 	}
 
@@ -989,8 +990,9 @@ void OptionEntryLanguageCode::CheckLanguagesAreInitialized() const
 	languages.emplace_back("uk", "Українська");
 
 	if (font_mpq) {
-		languages.emplace_back("zh_CN", "汉语");
-		languages.emplace_back("zh_TW", "漢語");
+		//remove chinese font
+		//languages.emplace_back("zh_CN", "汉语");
+		//languages.emplace_back("zh_TW", "漢語");
 	}
 
 	// Ensures that the ini specified language is present in languages list even if unknown (for example if someone starts to translate a new language)
diff --git a/Source/options.h b/Source/options.h
index 2ac3e33a..ab69aafa 100644
--- a/Source/options.h
+++ b/Source/options.h
@@ -569,7 +569,7 @@ struct KeymapperOptions : OptionCategoryBase {
 
 	void AddAction(
 	    string_view key, string_view name, string_view description, int defaultKey,
-	    std::function<void()> actionPressed, std::function<void()> actionReleased = nullptr, std::function<bool()> enable = nullptr, int index = -1);
+	    std::function<void()> actionPressed, std::function<void()> actionReleased = nullptr, std::function<bool()> enable = nullptr, int index = 0);
 	void KeyPressed(int key) const;
 	void KeyReleased(int key) const;
 	string_view KeyNameForAction(string_view actionName) const;
