diff --git a/Source/controls/plrctrls.cpp b/Source/controls/plrctrls.cpp
index 6d178d5a..7ad7126b 100644
--- a/Source/controls/plrctrls.cpp
+++ b/Source/controls/plrctrls.cpp
@@ -685,6 +685,7 @@ void ResetInvCursorPosition()
 	mousePos.y -= (InventorySlotSizeInPixels.height / 2);
 
 	SetCursorPos(mousePos);
+	MousePosition = mousePos;
 }
 
 int FindClosestInventorySlot(Point mousePos)
@@ -1010,6 +1011,7 @@ void InvMove(AxisDirection dir)
 	}
 
 	SetCursorPos(mousePos);
+	MousePosition = mousePos;
 }
 
 void HotSpellMove(AxisDirection dir)
@@ -1293,6 +1295,7 @@ void HandleRightStickMotion()
 		if (now - lastMouseSetTick > 0) {
 			SetCursorPos({ x, y });
 			lastMouseSetTick = now;
+			MousePosition = { x, y };
 		}
 	}
 }
