diff --git a/Source/controls/plrctrls.cpp b/Source/controls/plrctrls.cpp
index 6d178d5a..7ad7126b 100644
--- a/Source/controls/plrctrls.cpp
+++ b/Source/controls/plrctrls.cpp
@@ -641,6 +641,7 @@ void ResetInvCursorPosition()
 	mousePos.y -= (InventorySlotSizeInPixels.height / 2);
 
 	SetCursorPos(mousePos.x, mousePos.y);
+	MousePosition = mousePos;
 }
 
 /**
@@ -934,6 +935,7 @@ void InvMove(AxisDirection dir)
 	}
 
 	SetCursorPos(mousePos.x, mousePos.y);
+	MousePosition = mousePos;
 }
 
 /**
@@ -1284,6 +1286,7 @@ void HandleRightStickMotion()
 		if (now - lastMouseSetTick > 0) {
 			SetCursorPos(x, y);
 			lastMouseSetTick = now;
+			MousePosition = { x, y };
 		}
 	}
 }
