diff --git a/Source/options.cpp b/Source/options.cpp
index 2982587a..4426bf45 100644
--- a/Source/options.cpp
+++ b/Source/options.cpp
@@ -221,7 +221,7 @@ void SaveIni()
 #if SDL_VERSION_ATLEAST(2, 0, 0)
 bool HardwareCursorDefault()
 {
-#if defined(__ANDROID__) || defined(TARGET_OS_IPHONE)
+#if 1//#if defined(__ANDROID__) || defined(TARGET_OS_IPHONE)
 	// See https://github.com/diasurgical/devilutionX/issues/2502
 	return false;
 #else
@@ -306,7 +306,7 @@ Options sgOptions;
 #if SDL_VERSION_ATLEAST(2, 0, 0)
 bool HardwareCursorSupported()
 {
-#if defined(TARGET_OS_IPHONE)
+#if 1//#if defined(TARGET_OS_IPHONE)
 	return false;
 #else
 	SDL_version v;
@@ -953,7 +953,7 @@ void OptionEntryLanguageCode::LoadFromIni(string_view category)
 	}
 
 	LogVerbose("No suitable translation found");
-	strcpy(szCode, "en");
+	strcpy(szCode, "ko");	// default language to ko
 }
 void OptionEntryLanguageCode::SaveToIni(string_view category) const
 {
@@ -978,7 +978,8 @@ void OptionEntryLanguageCode::CheckLanguagesAreInitialized() const
 	languages.emplace_back("it", "Italiano");
 
 	if (font_mpq) {
-		languages.emplace_back("ja", "日本語");
+		//remove japanese font
+		//languages.emplace_back("ja", "日本語");
 		languages.emplace_back("ko", "한국어");
 	}
 
@@ -990,8 +991,9 @@ void OptionEntryLanguageCode::CheckLanguagesAreInitialized() const
 	languages.emplace_back("uk", "Українська");
 
 	if (font_mpq) {
-		languages.emplace_back("zh_CN", "汉语");
-		languages.emplace_back("zh_TW", "漢語");
+		//remove chinese font
+		//languages.emplace_back("zh_CN", "汉语");
+		//languages.emplace_back("zh_TW", "漢語");
 	}
 
 	// Ensures that the ini specified language is present in languages list even if unknown (for example if someone starts to translate a new language)
