diff --git a/Source/controls/plrctrls.cpp b/Source/controls/plrctrls.cpp
index 3a7d48fa..505eb333 100644
--- a/Source/controls/plrctrls.cpp
+++ b/Source/controls/plrctrls.cpp
@@ -698,6 +698,7 @@ void ResetInvCursorPosition()
 	mousePos.y -= (InventorySlotSizeInPixels.height / 2);
 
 	SetCursorPos(mousePos);
+	MousePosition = mousePos;
 }
 
 int FindClosestInventorySlot(Point mousePos)
@@ -1023,6 +1024,7 @@ void InvMove(AxisDirection dir)
 	}
 
 	SetCursorPos(mousePos);
+	MousePosition = mousePos;
 }
 
 void HotSpellMove(AxisDirection dir)
@@ -1399,6 +1401,7 @@ void HandleRightStickMotion()
 			ResetCursor();
 			SetCursorPos({ x, y });
 			lastMouseSetTick = now;
+			MousePosition = { x, y };
 		}
 	}
 }
