diff --git a/gptokeyb.cpp b/gptokeyb.cpp
index c636574..4c626ad 100755
--- a/gptokeyb.cpp
+++ b/gptokeyb.cpp
@@ -1026,7 +1026,7 @@ int main(int argc, char* argv[])
   }
 
   // Create fake input device (not needed in kill mode)
-  //if (!kill_mode) {  initialise device, now that kill mode will work with config & xbox modes
+  if (!kill_mode) {
     uinp_fd = open("/dev/uinput", O_WRONLY | O_NONBLOCK);
     if (uinp_fd < 0) {
       printf("Unable to open /dev/uinput\n");
@@ -1058,7 +1058,7 @@ int main(int argc, char* argv[])
       printf("Unable to create UINPUT device.");
       return -1;
     }
-  //}
+  }
 
   if (const char* db_file = SDL_getenv("SDL_GAMECONTROLLERCONFIG_FILE")) {
     SDL_GameControllerAddMappingsFromFile(db_file);
