diff --git a/engine/Makefile b/engine/Makefile
index da9bf1b5..46b0a85d 100644
--- a/engine/Makefile
+++ b/engine/Makefile
@@ -138,6 +138,7 @@ BUILD_PTHREAD   = 1
 BUILD_SDL_IO    = 1
 BUILD_VORBIS    = 1
 BUILD_WEBM      = 1
+BUILD_RG351_RUMBLE = 1
 BUILDING        = 1
 INCLUDES        = $(SDKPATH)/usr/include \
                   $(SDKPATH)/usr/include/SDL2
@@ -147,6 +148,52 @@ LIBRARIES       = $(SDKPATH)/usr/lib
 CFLAGS          += -Wno-error=format-overflow -Wno-error=implicit-function-declaration -Wno-error=unused-variable -Wno-error=unused-label -Wno-error=stringop-overflow
 endif
 
+ifdef BUILD_LINUX_x86_64
+TARGET 	        = $(VERSION_NAME).elf
+TARGET_FINAL    = $(VERSION_NAME)
+TARGET_PLATFORM = LINUX
+BUILD_SDL       = 1
+BUILD_GFX       = 1
+BUILD_PTHREAD   = 1
+BUILD_SDL_IO    = 1
+BUILD_OPENGL    = 1
+BUILD_LOADGL    = 1
+BUILD_VORBIS    = 1
+BUILD_WEBM      = 1
+BUILDING        = 1
+YASM 	          = nasm
+OBJTYPE         = elf
+INCLUDES        = $(SDKPATH)/usr/include \
+                  $(SDKPATH)/usr/include/SDL2
+TARGET_ARCH     = amd64
+BUILD_AMD64     = 1
+ARCHFLAGS       = -m64
+LIBRARIES       = $(SDKPATH)/usr/lib
+CFLAGS          += -DAMD64
+# Workaround for GCC 9
+CFLAGS          += -Wno-error=format-overflow -Wno-error=stringop-truncation
+endif
+
+ifdef BUILD_LINUX_aarch64
+TARGET 	        = $(VERSION_NAME).elf
+TARGET_FINAL    = $(VERSION_NAME)
+TARGET_PLATFORM = LINUX
+BUILD_LINUX     = 1
+BUILD_SDL       = 1
+BUILD_GFX       = 1
+BUILD_PTHREAD   = 1
+BUILD_SDL_IO    = 1
+BUILD_VORBIS    = 1
+BUILD_WEBM      = 1
+BUILDING        = 1
+INCLUDES        = $(SDKPATH)/usr/include \
+                  $(SDKPATH)/usr/include/SDL2
+OBJTYPE         = elf
+LIBRARIES       = $(SDKPATH)/usr/lib
+# Workaround for GCC 9
+CFLAGS          += -Wno-error=address -Wno-error=format-overflow -Wno-error=stringop-truncation -Wno-error=implicit-function-declaration -Wno-error=unused-variable -Wno-error=unused-label
+endif
+
 ifdef BUILD_DARWIN
 TARGET          = $(VERSION_NAME).elf
 TARGET_FINAL    = $(VERSION_NAME)
@@ -239,7 +286,7 @@ ifdef BUILD_WIN
 STRIP 	        = $(WINDEV)/$(PREFIX)strip$(EXTENSION) $(TARGET) -o $(TARGET_FINAL)
 endif
 ifdef BUILD_LINUX
-STRIP 	        = $(LNXDEV)/$(PREFIX)strip $(TARGET) -o $(TARGET_FINAL)
+STRIP 	        = $(PREFIX)-strip $(TARGET) -o $(TARGET_FINAL)
 endif
 ifdef BUILD_DARWIN
 STRIP           = $(PREFIX)strip $(TARGET) -o $(TARGET_FINAL)
@@ -593,6 +640,11 @@ CFLAGS 	       += -DWEBM
 endif
 
 
+ifdef BUILD_RG351_RUMBLE
+CFLAGS 	       += -DRG351_RUMBLE
+endif
+
+
 CXXFLAGS        = $(CFLAGS) -fno-exceptions -fno-rtti
 ASFLAGS         = $(CFLAGS)
 
diff --git a/engine/sdl/control.c b/engine/sdl/control.c
index 1a135de4..969783d7 100644
--- a/engine/sdl/control.c
+++ b/engine/sdl/control.c
@@ -980,6 +980,9 @@ void control_rumble(int port, int ratio, int msec)
 {
     #if SDL
     if (joystick[port] != NULL && joystick_haptic[port] != NULL) {
+        #ifdef RG351_RUMBLE
+        msec *= 10;
+        #endif
         if(SDL_HapticRumblePlay(joystick_haptic[port], ratio, msec) != 0)
         {
             //printf( "Warning: Unable to play rumble! %s\n", SDL_GetError() );
