diff --git a/engine/sdl/control.c b/engine/sdl/control.c
index 110e264d..394da482 100644
--- a/engine/sdl/control.c
+++ b/engine/sdl/control.c
         const char *buttonNames[] = {
             "A",
             "B",
@@ -799,6 +800,14 @@ const char *control_getkeyname(int deviceID, int keycode)
             "D-Pad Down",
             "D-Pad Left",
             "D-Pad Right",
+#if SDL_VERSION_ATLEAST(2, 0, 16)
+			"X share button",
+			"paddle P1",
+			"paddle P2",
+			"paddle P3",
+			"paddle P4",
+			"touchpad button",
+#endif
             "Left Stick Left",
             "Left Stick Right",
             "Left Stick Up",
