diff --git a/src/supertux/resources.cpp b/src/supertux/resources.cpp
index 4d033bfb9..c4fd7e219 100644
--- a/src/supertux/resources.cpp
+++ b/src/supertux/resources.cpp
@@ -27,6 +27,7 @@
 #include "video/font.hpp"
 #include "video/surface.hpp"
 #include "video/ttf_font.hpp"
+#include "util/gettext.hpp"

 std::unique_ptr<MouseCursor> Resources::mouse_cursor;

@@ -94,7 +95,7 @@ Resources::get_font_for_locale(const tinygettext::Language& locale)
 
   if(lang == "ne")
     return "fonts/Dekko-Regular.ttf";
-  if(lang == "cmn" || lang == "ja" || lang == "zh")
+  if(lang == "cmn" || lang == "ja" || lang == "ko" || lang == "zh")
     return "fonts/NotoSansCJKjp-Medium.otf";
   if(lang == "he")
     return "fonts/VarelaRound-Regular.ttf";

