diff --git a/es-core/src/resources/Font.cpp b/es-core/src/resources/Font.cpp
index 92852f7f..9c7ea7c4 100644
--- a/es-core/src/resources/Font.cpp
+++ b/es-core/src/resources/Font.cpp
@@ -284,10 +284,9 @@ std::vector<std::string> getFallbackFontPaths()
 
 	// batocera
 	const char* paths[] = {
+		":/DroidSansFallback.ttf",
 		":/DroidSansFallbackFull.ttf",// japanese, chinese, present on Debian
-		":/fontawesome_webfont.ttf",
 		":/fontawesome-webfont.ttf",
-		":/NanumMyeongjo.ttf", // korean font
 	};
 
 	std::vector<std::string> fontPaths;
