diff --git a/desktop_version/src/FileSystemUtils.cpp b/desktop_version/src/FileSystemUtils.cpp
index e9deec7..5bfa9da 100644
--- a/desktop_version/src/FileSystemUtils.cpp
+++ b/desktop_version/src/FileSystemUtils.cpp
@@ -1021,7 +1021,7 @@ bool FILESYSTEM_openDirectoryEnabled(void)
     return !SDL_GetHintBoolean("SteamTenfoot", SDL_FALSE);
 }
 
-#if defined(__EMSCRIPTEN__)
+#if !defined(_WIN32)
 bool FILESYSTEM_openDirectory(const char *dname)
 {
     return false;
diff --git a/desktop_version/src/UtilityClass.h b/desktop_version/src/UtilityClass.h
index 1f2a0f7..73c10f9 100644
--- a/desktop_version/src/UtilityClass.h
+++ b/desktop_version/src/UtilityClass.h
@@ -121,4 +121,8 @@ public:
 extern UtilityClass help;
 #endif
 
+#ifndef SDL_zeroa
+#define SDL_zeroa(x) SDL_memset((x), 0, sizeof((x)))
+#endif
+
 #endif /* UTILITYCLASS_H */
